cg/meta/upload/mutacc.py

```diff
@@ -0,0 +1,194 @@
+                            sample['individual_id'], case['_id'])
+                return False
+
+        return True
+
+    @staticmethod
+    def _has_causatives(case: dict) -> bool:
+        """
+            Check that the case has marked causative variants in scout
+
+            Args:
+                case (dict): case dictionary from scout
+
+            Returns:
+                (bool): True if case has marked causative variants in scout
+        """
+        if case.get('causatives'):
+            return True
+
+        LOG.warning("case %s has no marked causatives in scout", case['_id'])
+        return False
+
+
+# Reformat scout noutput to mutacc input
+
+MAPPER = namedtuple('mapper', ['field_name_1', 'field_name_2', 'conv'])
+
+def remap(input_dict: dict, mapper_list: list) -> dict:
+    """
+        Reformat dict from one application to be used by another
+
+        Args:
+            input_dict (dict): dictionary to be converted
+            mapper_list (list(MAPPER)): list of mapper objects
+
+        Returns:
+            output_dict (dict): conveted dictionary
+
+    """
+    output_dict = {}
+    for field in mapper_list:
+        if input_dict.get(field.field_name_1, None) is not None:
+            output_dict[field.field_name_2] = field.conv(input_dict[field.field_name_1])
+    return output_dict
+
+
+SCOUT_TO_MUTACC_SAMPLE = (
+    MAPPER('individual_id', 'sample_id', str),
+    MAPPER('sex', 'sex',
+           lambda sex: 'male' if sex == '1' else 'female' if sex == '2' else 'unknown'),
+    MAPPER('phenotype', 'phenotype', str),
+    MAPPER('father', 'father', lambda father: father if father else '0'),
+    MAPPER('mother', 'mother', lambda mother: mother if mother else '0'),
+    MAPPER('analysis_type', 'analysis_type', str),
+    MAPPER('bam_file', 'bam_file', str)
+)
+
+SCOUT_TO_MUTACC_CASE = (
+    MAPPER('_id', 'case_id', str),
+    MAPPER('genome_build', 'genome_build', str),
+    MAPPER('dynamic_gene_list', 'dynamic_gene_list', list),
+    MAPPER('panels', 'panels', lambda panels: [panel['panel_name'] for panel in panels]),
+    MAPPER('rank_model_version', 'rank_model_version', str),
+    MAPPER('rank_score_threshold', 'rank_score_threshold', int),
+    MAPPER('phenotype_terms', 'phenotype_terms', list),
+    MAPPER('phenotype_groups', 'phenotype_groups', list),
+    MAPPER('diagnosis_phenotypes', 'diagnosis_phenotypes', list),
+    MAPPER('diagnosis_genes', 'diagnosis_genes', list),
+    MAPPER('individuals', 'samples',
+           lambda samples: [remap(sample, SCOUT_TO_MUTACC_SAMPLE) for sample in samples])
+)
+
+def get_gene_string(genes):
+    """
+        Function to convert the 'genes' field in the scout variant document
+        to a string format that can be read by mutacc
+    """
+    gene_fields = ('hgnc_symbol',
+                   'region_annotation',
+                   'functional_annotation',
+                   'sift_prediction',
+                   'polyphen_prediction')
+
+    ann_info = []
```

Comment:
I think readability might be improved by writing out the full name?

Reply:
agree!